### ISSUE TITLE
**** MySQL server is not available. Waiting 5 seconds...

### ISSUE THREAD

**Issue by mik0s** [zabbix/zabbix-docker#974]
Hello. 

I'm trying to run zabbix using the docker-compose script suggested in the repository, but I can't run it.
```
zabbix-mysql-server-1             | 2022-06-15 16:59:49+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.29-1.el8 started.
zabbix-mysql-server-1             | 2022-06-15 16:59:49+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
zabbix-db_data_mysql-1 exited with code 0
zabbix-mysql-server-1             | 2022-06-15 16:59:49+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.29-1.el8 started.
zabbix-mysql-server-1             | '/var/lib/mysql/mysql.sock' -> '/var/run/mysqld/mysqld.sock'
zabbix-mysql-server-1             | 2022-06-15T16:59:50.002832Z 0 [Warning] [MY-010918] [Server] 'default_authentication_plugin' is deprecated and will be removed in a future release. Please use authentication_policy instead.
zabbix-mysql-server-1             | 2022-06-15T16:59:50.002893Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.29) starting as process 1
zabbix-mysql-server-1             | 2022-06-15T16:59:50.022577Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
zabbix-zabbix-server-1            | ** Preparing Zabbix server
zabbix-zabbix-server-1            | ** Using MYSQL_USER variable from ENV
zabbix-zabbix-server-1            | ** Using MYSQL_PASSWORD variable from ENV
zabbix-zabbix-server-1            | ** Using MYSQL_ROOT_PASSWORD variable from ENV
zabbix-zabbix-server-1            | ********************
zabbix-zabbix-server-1            | * DB_SERVER_HOST: mysql-server
zabbix-zabbix-server-1            | * DB_SERVER_PORT: 3306
zabbix-zabbix-server-1            | * DB_SERVER_DBNAME: zabbix
zabbix-zabbix-server-1            | ********************
zabbix-mysql-server-1             | 2022-06-15T16:59:50.427078Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
zabbix-mysql-server-1             | 2022-06-15T16:59:50.846156Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
zabbix-mysql-server-1             | 2022-06-15T16:59:50.846208Z 0 [System] [MY-013602] [Server] Channel mysql_main configured to support TLS. Encrypted connections are now supported for this channel.
zabbix-mysql-server-1             | 2022-06-15T16:59:50.871920Z 0 [System] [MY-010931] [Server] /usr/sbin/mysqld: ready for connections. Version: '8.0.29'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  MySQL Community Server - GPL.
zabbix-mysql-server-1             | 2022-06-15T16:59:50.871989Z 0 [System] [MY-011323] [Server] X Plugin ready for connections. Bind-address: '::' port: 33060, socket: /var/run/mysqld/mysqlx.sock
zabbix-zabbix-web-apache-mysql-1  | ** Deploying Zabbix web-interface (Apache) with MySQL database
zabbix-zabbix-web-apache-mysql-1  | ** Using MYSQL_USER variable from ENV
zabbix-zabbix-web-apache-mysql-1  | ** Using MYSQL_PASSWORD variable from ENV
zabbix-zabbix-web-apache-mysql-1  | ********************
zabbix-zabbix-web-apache-mysql-1  | * DB_SERVER_HOST: mysql-server
zabbix-zabbix-web-apache-mysql-1  | * DB_SERVER_PORT: 3306
zabbix-zabbix-web-apache-mysql-1  | * DB_SERVER_DBNAME: zabbix
zabbix-zabbix-web-apache-mysql-1  | ********************
zabbix-zabbix-server-1            | **** MySQL server is not available. Waiting 5 seconds...
zabbix-zabbix-web-apache-mysql-1  | **** MySQL server is not available. Waiting 5 seconds...
zabbix-zabbix-server-1            | **** MySQL server is not available. Waiting 5 seconds...
zabbix-zabbix-web-apache-mysql-1  | **** MySQL server is not available. Waiting 5 seconds...
zabbix-zabbix-server-1            | **** MySQL server is not available. Waiting 5 seconds...
zabbix-zabbix-web-apache-mysql-1  | **** MySQL server is not available. Waiting 5 seconds...
^CGracefully stopping... (press Ctrl+C again to force)
```

I am using Oracle Server and docker-compose file docker-compose_v3_ol_mysql_latest.yaml
I removed everything superfluous from it and left only zabbix-server, zabbix-mysql and zabbix-web-apache-mysql.
```
version: '3.5'
services:
 zabbix-server:
  image: zabbix/zabbix-server-mysql:ol-6.0-latest
  ports:
   - "10051:10051"
  volumes:
   - /etc/localtime:/etc/localtime:ro
   - /etc/timezone:/etc/timezone:ro
   - ./zbx_env/usr/lib/zabbix/alertscripts:/usr/lib/zabbix/alertscripts:ro
   - ./zbx_env/usr/lib/zabbix/externalscripts:/usr/lib/zabbix/externalscripts:ro
   - ./zbx_env/var/lib/zabbix/export:/var/lib/zabbix/export:rw
   - ./zbx_env/var/lib/zabbix/modules:/var/lib/zabbix/modules:ro
   - ./zbx_env/var/lib/zabbix/enc:/var/lib/zabbix/enc:ro
   - ./zbx_env/var/lib/zabbix/ssh_keys:/var/lib/zabbix/ssh_keys:ro
   - ./zbx_env/var/lib/zabbix/mibs:/var/lib/zabbix/mibs:ro
   - snmptraps:/var/lib/zabbix/snmptraps:rw
  ulimits:
   nproc: 65535
   nofile:
    soft: 20000
    hard: 40000
  deploy:
   resources:
    limits:
      cpus: '0.70'
      memory: 1G
    reservations:
      cpus: '0.5'
      memory: 512M
  env_file:
   - ./env_vars/.env_db_mysql
   - ./env_vars/.env_srv
  secrets:
   - MYSQL_USER
   - MYSQL_PASSWORD
   - MYSQL_ROOT_USER
   - MYSQL_ROOT_PASSWORD
#   - client-key.pem
#   - client-cert.pem
#   - root-ca.pem
  depends_on:
   - mysql-server
  networks:
   zbx_net_backend:
     aliases:
      - zabbix-server
      - zabbix-server-mysql
      - zabbix-server-ol-mysql
      - zabbix-server-mysql-ol
   zbx_net_frontend:
#  devices:
#   - "/dev/ttyUSB0:/dev/ttyUSB0"
  stop_grace_period: 30s
  sysctls:
   - net.ipv4.ip_local_port_range=1024 65000
   - net.ipv4.conf.all.accept_redirects=0
   - net.ipv4.conf.all.secure_redirects=0
   - net.ipv4.conf.all.send_redirects=0
  labels:
   com.zabbix.description: "Zabbix server with MySQL database support"
   com.zabbix.company: "Zabbix LLC"
   com.zabbix.component: "zabbix-server"
   com.zabbix.dbtype: "mysql"
   com.zabbix.os: "oracle linux"

 zabbix-web-apache-mysql:
  image: zabbix/zabbix-web-apache-mysql:ol-6.0-latest
  volumes:
   - /etc/localtime:/etc/localtime:ro
   - /etc/timezone:/etc/timezone:ro
   - ./zbx_env/etc/ssl/apache2:/etc/ssl/apache2:ro
   - ./zbx_env/usr/share/zabbix/modules/:/usr/share/zabbix/modules/:ro
  deploy:
   resources:
    limits:
      cpus: '0.70'
      memory: 512M
    reservations:
      cpus: '0.5'
      memory: 256M
  env_file:
   - ./env_vars/.env_db_mysql
   - ./env_vars/.env_web
  secrets:
   - MYSQL_USER
   - MYSQL_PASSWORD
#   - client-key.pem
#   - client-cert.pem
#   - root-ca.pem
  depends_on:
   - mysql-server
   - zabbix-server
  healthcheck:
   test: ["CMD", "curl", "-f", "http://localhost:8080/"]
   interval: 10s
   timeout: 5s
   retries: 3
   start_period: 30s
  networks:
   zbx_net_backend:
    aliases:
     - zabbix-web-apache-mysql
     - zabbix-web-apache-ol-mysql
     - zabbix-web-apache-mysql-ol
   zbx_net_frontend:
   web:
  stop_grace_period: 10s
  sysctls:
   - net.core.somaxconn=65535
  labels:
   com.zabbix.description: "Zabbix frontend on Apache web-server with MySQL database support"
   com.zabbix.company: "Zabbix LLC"
   com.zabbix.component: "zabbix-frontend"
   com.zabbix.webserver: "apache2"
   com.zabbix.dbtype: "mysql"
   com.zabbix.os: "oracle linux"
   traefik.backend: "zabbix"
   traefik.frontend.rule: "Host:zabbix.mydomain.local"
   traefik.docker.network: "web"
   traefik.port: "8080"

 zabbix-agent:
  image: zabbix/zabbix-agent:ol-6.0-latest
  profiles:
   - full
   - all
  ports:
   - "10050:10050"
  volumes:
   - /etc/localtime:/etc/localtime:ro
   - /etc/timezone:/etc/timezone:ro
   - ./zbx_env/etc/zabbix/zabbix_agentd.d:/etc/zabbix/zabbix_agentd.d:ro
   - ./zbx_env/var/lib/zabbix/modules:/var/lib/zabbix/modules:ro
   - ./zbx_env/var/lib/zabbix/enc:/var/lib/zabbix/enc:ro
   - ./zbx_env/var/lib/zabbix/ssh_keys:/var/lib/zabbix/ssh_keys:ro
  deploy:
   resources:
    limits:
      cpus: '0.2'
      memory: 128M
    reservations:
      cpus: '0.1'
      memory: 64M
   mode: global
  env_file:
   - ./env_vars/.env_agent
  privileged: true
  pid: "host"
  networks:
   zbx_net_backend:
    aliases:
     - zabbix-agent
     - zabbix-agent-passive
     - zabbix-agent-ol
  stop_grace_period: 5s
  labels:
   com.zabbix.description: "Zabbix agent"
   com.zabbix.company: "Zabbix LLC"
   com.zabbix.component: "zabbix-agentd"
   com.zabbix.os: "oracle linux"

 mysql-server:
  image: mysql:8.0-oracle
  command:
   - mysqld
   - --character-set-server=utf8mb4
   - --collation-server=utf8mb4_bin
   - --default-authentication-plugin=mysql_native_password
#   - --require-secure-transport
#   - --ssl-ca=/run/secrets/root-ca.pem
#   - --ssl-cert=/run/secrets/server-cert.pem
#   - --ssl-key=/run/secrets/server-key.pem
  volumes:
   - ./zbx_env/var/lib/mysql:/var/lib/mysql:rw
  env_file:
   - ./env_vars/.env_db_mysql
  secrets:
   - MYSQL_USER
   - MYSQL_PASSWORD
   - MYSQL_ROOT_PASSWORD
#   - server-key.pem
#   - server-cert.pem
#   - root-ca.pem
  stop_grace_period: 1m
  networks:
   zbx_net_backend:
    aliases:
     - mysql-server
     - zabbix-database
     - mysql-database

 db_data_mysql:
  image: busybox
  volumes:
   - ./zbx_env/var/lib/mysql:/var/lib/mysql:rw

# elasticsearch:
#  image: elasticsearch
#  profiles:
#   - full
#   - all
#  environment:
#   - transport.host=0.0.0.0
#   - discovery.zen.minimum_master_nodes=1
#  networks:
#   zbx_net_backend:
#    aliases:
#     - elasticsearch

networks:
  zbx_net_frontend:
    driver: bridge
    driver_opts:
      com.docker.network.enable_ipv6: "false"
    ipam:
      driver: default
      config:
      - subnet: 172.16.238.0/24
  zbx_net_backend:
    driver: bridge
    driver_opts:
      com.docker.network.enable_ipv6: "false"
    internal: true
    ipam:
      driver: default
      config:
      - subnet: 172.16.239.0/24
  web:
    external: true

volumes:
  snmptraps:

secrets:
  MYSQL_USER:
    file: ./env_vars/.MYSQL_USER
  MYSQL_PASSWORD:
    file: ./env_vars/.MYSQL_PASSWORD
  MYSQL_ROOT_USER:
    file: ./env_vars/.MYSQL_ROOT_USER
  MYSQL_ROOT_PASSWORD:
    file: ./env_vars/.MYSQL_ROOT_PASSWORD
#  client-key.pem:
#    file: ./env_vars/.ZBX_DB_KEY_FILE
#  client-cert.pem:
#    file: ./env_vars/.ZBX_DB_CERT_FILE
#  root-ca.pem:
#    file: ./env_vars/.ZBX_DB_CA_FILE
#  server-cert.pem:
#    file: ./env_vars/.DB_CERT_FILE
#  server-key.pem:
#    file: ./env_vars/.DB_KEY_FILE
```

From the mysql-server container I can connect to the mysql server, but from the zabbix-server container I can't.
```
# docker exec -ti 335ddd90be12 bash
bash-4.4$ ping mysql-server
PING mysql-server (172.16.239.2) 56(84) bytes of data.
64 bytes from zabbix-mysql-server-1.zabbix_zbx_net_backend (172.16.239.2): icmp_seq=1 ttl=64 time=0.210 ms
64 bytes from zabbix-mysql-server-1.zabbix_zbx_net_backend (172.16.239.2): icmp_seq=2 ttl=64 time=0.150 ms
^C
--- mysql-server ping statistics ---
2 packets transmitted, 2 received, 0% packet loss, time 1002ms
rtt min/avg/max/mdev = 0.150/0.180/0.210/0.030 ms

bash-4.4$ mysql -h mysql-server -u root -proot_pwd zabbix
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 2003 (HY000): Can't connect to MySQL server on 'mysql-server:3306' (113)
```

My envelopment
```
# docker version
Client: Docker Engine - Community
 Version:           20.10.17
 API version:       1.41
 Go version:        go1.17.11
 Git commit:        100c701
 Built:             Mon Jun  6 23:03:11 2022
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.17
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.17.11
  Git commit:       a89b842
  Built:            Mon Jun  6 23:01:29 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.6
  GitCommit:        10c12954828e7c7c9b6e0ea9b0c02b01407d3ae1
 runc:
  Version:          1.1.2
  GitCommit:        v1.1.2-0-ga916309
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

# docker-compose version
Docker Compose version v2.6.0
```

PS: I also tried using postgres - the result is the same.

**Comment by dotneft** [comment]
is 172.16.239.2 IP of MySQL container? Are you able to connect to MySQL locally from mysql container?

**Comment by mik0s** [comment]
Yes. From container MySQL-server I can connect. And I can connect from zabbix-server if set same network for mysql and zabbix container.

**Comment by dotneft** [comment]
what do you mean the same network?

**Comment by mik0s** [comment]
I added an external network "web" and specified it for both services.  in this case they see each other.

**Comment by democles9308** [comment]
Hi, I've got the same problem... have been digging a  little bit and it seems the root user has no permission to access the  Mysql server, hence the error message *** MySQL server is not available. Waiting 5 seconds ****
If you put  DEBUG_MODE=True in the .env_db_mysql file and restart you get some additional  info. To test the connection the following is executed:
`++ mysqladmin ping -h mysql-server -P 3306 -u root --silent --connect_timeout=10
+ '[' '!' '' ']'
+ echo '**** MySQL server is not available. Waiting 5 seconds...'`

If you go into the  zabbix-container and execute the same command  you get:
`bash-5.1$ mysqladmin ping -h mysql-server -P 3306 -u root -p --connect_timeout=10
Enter password: 
mysqladmin: connect to server at 'mysql-server' failed
error: 'Access denied for user 'root'@'172.16.239.3' (using password: YES)'
bash-5.1$ mysqladmin ping -h mysql-server -P 3306 -u root
mysqladmin: connect to server at 'mysql-server' failed
error: 'Access denied for user 'root'@'172.16.239.3' (using password: NO)'`

When using zabbix as user :
`bash-5.1$ mysqladmin ping -h mysql-server -P 3306 -u zabbix -p  --connect_timeout=10
Enter password: 
mysqld is alive`

So the connection between Zabbix-server and Mysql server is ok as they both are part of the ZBX_NET_BACKEND..
When I open the site I got the following  error:
Database error : Unable to determine current Zabbix database version: the table "dbversion" was not found.

 Hope this can help (a bit)
